### PR TITLE
feat: source level filter support

### LINKED ISSUE #1264: source level filter support
## Feature Request

**Is your feature request related to a problem? Please describe:**

run many tasks for the same MySQL source and an incompatible DDL applied in the source.

**Describe the feature you'd like:**

now, we need to add a filter rule for this incompatible DDL in all tasks' config.

but if we can only add a filter for all (sub)tasks with this source, it's easier to use.

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

### PR DESCRIPTION
**Pull request by GMHDBJD**
### What problem does this PR solve? 
close #1264 

### What is changed and how it works?
add binlog filter config in source configuartion file.

### Check List 

Tests 

 - Integration test

### CODE REVIEW

**Comment by lance6716** [comment]
test failed because unrevealed bug in #1366 😂 I will file a small PR to fix

```
[2021-01-12T02:13:19.192Z] FAIL: check_test.go:87: testCheckerSuite.TestDumpPrivilegeChecking
[2021-01-12T02:13:19.192Z] 
[2021-01-12T02:13:19.192Z] check_test.go:96:
[2021-01-12T02:13:19.192Z]     c.Assert(CheckSyncConfig(context.Background(), cfgs), tc.ErrorMatches, "(.|\n)*lack of RELOAD,Select privilege(.|\n)*")
[2021-01-12T02:13:19.192Z] ... error string = "" +
[2021-01-12T02:13:19.192Z] ...     "[code=26005:class=task-check:scope=internal:level=medium], Message: fail to check synchronization configuration with type: check was failed, please see detail\n" +
[2021-01-12T02:13:19.192Z] ...     " detail: {\n" +
[2021-01-12T02:13:19.192Z] ...     "\t\t\"results\": [\n" +
[2021-01-12T02:13:19.192Z] ...     "\t\t\t{\n" +
[2021-01-12T02:13:19.192Z] ...     "\t\t\t\t\"id\": 0,\n" +
[2021-01-12T02:13:19.192Z] ...     "\t\t\t\t\"name\": \"source db dump privilege checker\",\n" +
[2021-01-12T02:13:19.192Z] ...     "\t\t\t\t\"desc\": \"check dump privileges of source DB\",\n" +
[2021-01-12T02:13:19.192Z] ...     "\t\t\t\t\"state\": \"fail\",\n" +
[2021-01-12T02:13:19.192Z] ...     "\t\t\t\t\"errorMsg\": \"lack of Select,RELOAD privilege\",\n" +
[2021-01-12T02:13:19.192Z] ...     "\t\t\t\t\"instruction\": \"GRANT Select,RELOAD ON *.* TO 'haha'@'%';\",\n" +
[2021-01-12T02:13:19.192Z] ...     "\t\t\t\t\"extra\": \"address of db instance - :0\"\n" +
[2021-01-12T02:13:19.192Z] ...     "\t\t\t}\n" +
[2021-01-12T02:13:19.192Z] ...     "\t\t],\n" +
[2021-01-12T02:13:19.192Z] ...     "\t\t\"summary\": {\n" +
[2021-01-12T02:13:19.192Z] ...     "\t\t\t\"passed\": false,\n" +
[2021-01-12T02:13:19.192Z] ...     "\t\t\t\"total\": 1,\n" +
[2021-01-12T02:13:19.192Z] ...     "\t\t\t\"successful\": 0,\n" +
[2021-01-12T02:13:19.192Z] ...     "\t\t\t\"failed\": 1,\n" +
[2021-01-12T02:13:19.192Z] ...     "\t\t\t\"warning\": 0\n" +
[2021-01-12T02:13:19.192Z] ...     "\t\t}\n" +
[2021-01-12T02:13:19.192Z] ...     "\t}"
[2021-01-12T02:13:19.192Z] ... regex string = "" +
[2021-01-12T02:13:19.192Z] ...     "(.|\n" +
[2021-01-12T02:13:19.192Z] ...     ")*lack of RELOAD,Select privilege(.|\n" +
[2021-01-12T02:13:19.192Z] ...     ")*"
[2021-01-12T02:13:19.192Z] 
[2021-01-12T02:13:19.192Z] OOPS: 11 passed, 1 FAILED
```

File: dm/worker/worker.go [hunk 1]
```diff
@@ -634,6 +639,25 @@ func copyConfigFromSource(cfg *config.SubTaskConfig, sourceCfg *config.SourceCon
 
 	// we can remove this from SubTaskConfig later, because syncer will always read from relay
 	cfg.AutoFixGTID = sourceCfg.AutoFixGTID
+
+	cfg.CaseSensitive = cfg.CaseSensitive || sourceCfg.CaseSensitive
```
Comment (lance6716) [inline dm/worker/worker.go:643]: How about warning here

**Review by lance6716 — COMMENTED** [review]
rest LGTM

**Review by leoppro — APPROVED** [review]

**Pull request merged**